DAL/EmployeeDAL.cs

```diff
@@ -0,0 +1,218 @@
+        Employee employeeToUpdate = existingEmployees.FirstOrDefault(emp => emp.EmpNo == empNo);
+
+        if (employeeToUpdate == null)
+        {
+            return false;
+        }
+
+        // Update employee fields
+        employeeToUpdate.FirstName = updatedEmployee.FirstName == "--d" ? null :
+                                     string.IsNullOrWhiteSpace(updatedEmployee.FirstName) ? employeeToUpdate.FirstName :
+                                     updatedEmployee.FirstName;
+        employeeToUpdate.LastName = updatedEmployee.LastName == "--d" ? null :
+                                    string.IsNullOrWhiteSpace(updatedEmployee.LastName) ? employeeToUpdate.LastName :
+                                    updatedEmployee.LastName;
+        employeeToUpdate.Dob = updatedEmployee.Dob == DateTime.MinValue ? null :
+                            updatedEmployee.Dob ?? employeeToUpdate.Dob;
+        employeeToUpdate.Email = updatedEmployee.Email == "--d" ? null :
+                                 string.IsNullOrWhiteSpace(updatedEmployee.Email) ? employeeToUpdate.Email :
+                                 updatedEmployee.Email;
+        employeeToUpdate.MobileNumber = updatedEmployee.MobileNumber == "--d" ? null :
+                                         string.IsNullOrWhiteSpace(updatedEmployee.MobileNumber) ? employeeToUpdate.MobileNumber :
+                                         updatedEmployee.MobileNumber;
+        employeeToUpdate.JoiningDate = updatedEmployee.JoiningDate == DateTime.MinValue ? null :
+                                        updatedEmployee.JoiningDate ?? updatedEmployee.JoiningDate;
+        employeeToUpdate.LocationId = updatedEmployee.LocationId == -1 ? null :
+                                   updatedEmployee.LocationId ?? employeeToUpdate.LocationId;
+        employeeToUpdate.JobTitle = updatedEmployee.JobTitle == "--d" ? null :
+                                    string.IsNullOrWhiteSpace(updatedEmployee.JobTitle) ? employeeToUpdate.JobTitle :
+                                    updatedEmployee.JobTitle;
+        employeeToUpdate.DepartmentId = updatedEmployee.DepartmentId == -1 ? null :
+                                     updatedEmployee.DepartmentId ?? employeeToUpdate.DepartmentId;
+        employeeToUpdate.AssignManager = updatedEmployee.AssignManager == "--d" ? null :
+                                          string.IsNullOrWhiteSpace(updatedEmployee.AssignManager) ? employeeToUpdate.AssignManager :
+                                          updatedEmployee.AssignManager;
+        employeeToUpdate.AssignProject = updatedEmployee.AssignProject == "--d" ? null :
+                                          string.IsNullOrWhiteSpace(updatedEmployee.AssignProject) ? employeeToUpdate.AssignProject :
+                                          updatedEmployee.AssignProject;
+
+        int index = existingEmployees.FindIndex(emp => emp.EmpNo == empNo);
```

Comment:
You are already finding initially right. Finding multiple times is not a good idea.